tests/subsys/net/lib/nrf_cloud/src/main.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <ztest.h>
+#include <zephyr.h>
+#include <stdio.h>
+#include <net/nrf_cloud.h>
+#include "nrf_cloud_fsm.h"
+
+/* Main test function that declares a test suite
+*  includes all the test cases implemented below
+*/
+ZTEST_SUITE(nrf_cloud_init_test, NULL, NULL, NULL, NULL, NULL);
+
+/* The library can only be initialized when the internal state is
+*  not idle, so the test verifies that nrf_loud_init function will
+*  return with permission denied error (-EACCES) with said condition
+*/
+ZTEST(nrf_cloud_init_test, test_init_not_idle)
+{
+	struct nrf_cloud_init_param params = {
+		.event_handler = NULL,
+		.client_id = NULL,
+	};
+
+	/* The function is an interface (declared in nrf_cloud_fsm.h)
+	*  and implemented inside nrf_cloud.c, so mocking would not be
+	*  a good idea. Here it changes the internal state to not idle
+	*/
```

Comment:
I don't think you need to talk directly about the decision not to mock the `fsm`,

Maybe phrase this as:

> nfsm_set_current_state_and_notify directly sets the internal state of the nrf_cloud lib, allowing us to easily test various initialization scenarios.